src/aura/BGE_BatchGiftEntryTab/BGE_BatchGiftEntryTabHelper.js

```diff
@@ -1,50 +1,75 @@
 ({
+    /******************************** Init Functions *****************************/
+
     doInit: function(component) {
-        var action = component.get('c.getNamespacedListView');
+        var action = component.get('c.getTabModel');
         action.setCallback(this, function (response) {
             const state = response.getState();
             if (state === 'SUCCESS') {
-                let namespacedListView = response.getReturnValue();
-                $A.createComponent(
-                    "lightning:listView",
-                    {
-                        "objectApiName": namespacedListView.objectApiName,
-                        "listName": namespacedListView.listName,
-                        "showActionBar": false,
-                        "enableInlineEdit": false,
-                        "showRowLevelActions": false
-                    },
-                    function (listView, status, errorMessage) {
-                        if (status === "SUCCESS") {
-                            let body = component.get("v.body");
-                            body.push(listView);
-                            component.set("v.body", body);
-                        } else {
-                            this.showToast(component, $A.get('$Label.c.PageMessagesError'), errorMessage, 'error');
-                        }
-                    }
-                );
+                let model = JSON.parse(response.getReturnValue());
+                this.setColumns(component, model.columns);
+                component.set('v.totalNumberOfRows', model.totalNumberOfRows);
```

Comment:
Might make sense to break this out into a new function, especially if we need to query for it at regular intervals per Patrick's suggestion?